### PR TITLE
Feature/baseline mods

### PR DESCRIPTION
**Pull request by jbrdge**
# PR: Wire Baseline into Make/Compose, Docs & Changelog (Modifications-only)

**Type:** Integration / DX improvements  
**Depends on:** Additions-only PR (baseline CLI + scripts)  
**Scope:** Modify tracked files only (no new untracked additions).

---

## Summary

Connect the baseline CLI and scripts into the Docker-first workflow, improve developer experience, and update docs:

- **Makefile**: add targets `make-features-from-archive`, `train-baseline`, `train-baseline-sample`, `show-metrics`, `ls-artifacts`, `monte-carlo`, `monte-carlo-summary`, `mc-best`, `mc-show-best`, `mc-show-best-metrics`, `mc-ls-best`. Avoid heredocs; use `--entrypoint sh -c`.
- **docker-compose.yml**: optionally run as host user to avoid root-owned artifacts (`user: "${HOST_UID:-1000}:${HOST_GID:-1000}"`); set headless plotting env (`MPLBACKEND=Agg`, `MPLCONFIGDIR=/tmp/matplotlib`, `XDG_CACHE_HOME=/tmp`).
- **requirements.txt**: pin core libs for reproducibility.
- **README.md**: add Baseline Model section (quickstart, artifacts, sampling, Monte Carlo, best-run helpers, permissions note).
- **CHANGELOG.md**: add `[0.5.0]` entry with concise bullets.
- **.env.example**: include `HOST_UID/HOST_GID` defaults (optional note).
- **src/db.py**: clarify/leave as placeholder or remove if appropriate.

---

## Acceptance Criteria

- [ ] `make make-features-from-archive [N=…]` produces `data/processed/features.parquet` from Kaggle CSV.
- [ ] `make train-baseline` finishes and writes all artifacts under `artifacts/baseline_v1/`.
- [ ] `make show-metrics` pretty-prints metrics; `make ls-artifacts` lists artifacts.
- [ ] `make monte-carlo N=50 MC_ITERS=3` produces `artifacts/mc_baseline/metrics.csv` and per-run artifacts.
- [ ] `make mc-best [METRIC=f1|roc_auc]` copies best run to `artifacts/mc_baseline/best/` with `best_summary.json`.
- [ ] Plots render (no blank images) in headless containers.
- [ ] Files written by containers are owned by the host user on Linux/WSL when `HOST_UID/GID` is set.
- [ ] README/CHANGELOG updated accordingly.

---

## Tasks

- [ ] Update Makefile with all new targets (no heredocs; container `--entrypoint sh -c`)
- [ ] Update docker-compose.yml with optional `user:` and plotting env vars
- [ ] Add `scripts/docker-env.sh` and reference it in README
- [ ] Pin versions in requirements.txt (pandas, scikit-learn, matplotlib, pyarrow, joblib)
- [ ] Add README Baseline section + Monte Carlo + mc-best helpers + permissions
- [ ] Update CHANGELOG `[0.5.0]`
- [ ] (Optional) Add `.gitkeep` under `artifacts/` and `data/processed/`
- [ ] Decide on `src/db.py` (placeholder comment or remove)

---

## How to Test (integration, via Make)

```bash
# Linux/WSL only (set host UID/GID for correct file ownership)
source scripts/docker-env.sh

# Build features (optionally sample N rows)
make make-features-from-archive N=500

# Train baseline model
make train-baseline

# Show metrics
make show-metrics

# Monte Carlo experiment (3 runs, 50 samples each)
make monte-carlo N=50 MC_ITERS=3

# Pick best run by ROC-AUC
make mc-best METRIC=roc_auc
make mc-show-best
```

---

## Out of Scope

- New modeling algorithms / hyperparameter tuning
- DB schema or ETL logic changes
- CI pipeline (can be follow-up)

---

## Risks & Mitigations

- **Permissions variance across dev machines:**  
  Use `${HOST_UID}:${HOST_GID}` mapping and document in README. Provide `scripts/docker-env.sh` helper.

- **Matplotlib headless quirks:**  
  Set `MPLBACKEND=Agg` and writable cache dirs via env; use explicit figure creation in CLI.

- **Makefile fragility:**  
  Avoid heredocs; keep commands on single lines; rely on small scripts in `scripts/`.

### CODE REVIEW

**Review by jbreckenridge — CHANGES_REQUESTED** [review]
Noticed that make make-features-from-archive N=5000 still writes the full 7,043 rows. Looks like the --sample flag isn’t being applied correctly. This needs changes to ensure the row cap works as expected.

**Review by jbreckenridge — APPROVED** [review]
Approved, fixes resolved issue with samples. Tested successfully.

**Comment by jbreckenridge** [comment]
Closes #42
